FormalConjectures/Wikipedia/SumOfThreeCubes.lean

```diff
@@ -32,6 +32,10 @@ asked by user [*David Feldman*](https://mathoverflow.net/users/10909/david-feldm
 def IsSumOfThreeCubes (n : ℤ) : Prop :=
   ∃ x y z : ℤ, n = x^3 + y^3 + z^3
 
+/-- The predicate that `n : ℚ` is a sum of three *rational* cubes. -/
+def IsSumOfThreeCubesRat (n : ℚ) : Prop :=
+  ∃ x y z : ℚ, n = x^3 + y^3 + z^3
+
```

Comment:
I would suggest to generalize here instead of copying the definition:
```
variable {R : Type*} [Ring R]

/-- The predicate that `n : R` is a sum of three cubes. -/
def IsSumOfThreeCubes (n : R) : Prop :=
  ∃ x y z : R, n = x^3 + y^3 + z^3
```
Then one needs to explicitely give the type below: replace `2` by `(2 : ℤ) ` etc.
```
theorem isSumOfThreeCubes_2 : IsSumOfThreeCubes (2 : ℤ) :=

``